Reduce 6.0 release branch CI runs (6.0 preview 2)

BruceForstall opened the pull request:
6.0 release branches are running AzDO stress pipelines on a schedule, even though (normally) nothing is changing. Change this to only run if something changes.

This is a port of https://github.com/dotnet/runtime/pull/49020 to the release branch.

BruceForstall committed:
Increase jitstress job run frequency

Change #41856 reduced stress job frequency to once per day, alternating
between main and the release branches. However, the release branches
get few changes yet still run the job. (Also, all the release branches
run the job at the same time.)

Change to running the job daily in main, where most development
happens. Also change the runs to daily in the release branches, but
change to only run if the branch has changed since the last run.

hoyosjs commented on eng/pipelines/coreclr/jitstress.yml:
```diff
@@ -1,18 +1,18 @@
+  displayName: Daily at 8:00 PM (UTC-8:00)
   branches:
     include:
     - master
```
Master no longer exists, I'd revert these lines of the change as this will only after the branch you check it in.

BruceForstall replied:
I suppose we could just delete the "master" section of this (and other) yml files, since they're unused in the release branches anyway.

I did this just as a "cherry-pick" from the 'main' branch change, so it doesn't really hurt, either

hoyosjs reviewed (approved):
Just a small nit.

mmitche commented:
We should hold off on this. I believe the last runtime change for p2 has left the station

BruceForstall commented:
> We should hold off on this. I believe the last runtime change for p2 has left the station

@mmitche the issue here is that this is an infra-only change. The 6.0p1 and 6.0p2 branches continue to run these pipelines 4 times per week, and I want them to stop (unless needed).

mmitche commented:
Gotcha. Even if there are no changes?

Anyways, please cancel the official build once this is merged.

BruceForstall commented:
e.g., look at all the "6.0-preview1" and "6.0-preview2" runs here: https://dev.azure.com/dnceng/public/_build?definitionId=658

BruceForstall commented:
Yes, the pipelines are running even if there are no changes (now).

> Anyways, please cancel the official build once this is merged.

You'll need to tell me how that is done. (does the official build automatically get kicked off for every merge?)

BruceForstall commented:
@ViktorHofer @hoyosjs @safern This change (and the related 6.0 preview 1 change: https://github.com/dotnet/runtime/pull/49168) didn't work. I change the YAML to use "always: false" for the schedule, meaning it should only trigger if there is a change. However, we're still getting test runs every day, and they all have a single change. Technically, the documentation says, "if there have been source code changes since the last successful scheduled run". Since there hasn't been a successful run, that's probably why we're getting runs.

Given that we don't care about these runs, I propose to push new PRs to remove the schedules entirely for the preview releases, as well as for main (before it gets branched again).

Comments?
